src/webview.rs

```diff
@@ -1,271 +1,57 @@
-                        (position.x, position.y, WheelMode::DeltaPixel)
-                    }
-                };
-
-                // Wheel Event
-                events.push(EmbedderEvent::Wheel(
-                    WheelDelta { x, y, z: 0.0, mode },
-                    DevicePoint::new(
-                        self.mouse_position.get().x as f32,
-                        self.mouse_position.get().y as f32,
-                    ),
-                ));
-
-                // Scroll Event
-                // Do one axis at a time.
-                if y.abs() >= x.abs() {
-                    x = 0.0;
-                } else {
-                    y = 0.0;
-                }
-
-                let phase: TouchEventType = match phase {
-                    TouchPhase::Started => TouchEventType::Down,
-                    TouchPhase::Moved => TouchEventType::Move,
-                    TouchPhase::Ended => TouchEventType::Up,
-                    TouchPhase::Cancelled => TouchEventType::Cancel,
-                };
-
-                events.push(EmbedderEvent::Scroll(
-                    ScrollLocation::Delta(LayoutVector2D::new(x as f32, y as f32)),
-                    DeviceIntPoint::new(
-                        self.mouse_position.get().x as i32,
-                        self.mouse_position.get().y as i32,
-                    ),
-                    phase,
-                ));
-            }
-            WindowEvent::CloseRequested => {
-                events.push(EmbedderEvent::Quit);
-            }
-            e => log::warn!("Verso hasn't supported this window event yet: {e:?}"),
-        }
+    /// Set the history URLs and current index of the webview
+    pub fn set_history(&self, history: Vec<ServoUrl>, current: usize) {
+        self.history.replace(history);
+        self.current.replace(current);
     }
 }
 
-unsafe impl Send for WebView {}
-unsafe impl Sync for WebView {}
-
-impl WindowMethods for WebView {
-    fn get_coordinates(&self) -> EmbedderCoordinates {
-        let size = self.window.inner_size();
-        let pos = Point2D::new(0, 0);
-        let viewport = Size2D::new(size.width as i32, size.height as i32);
+/// A panel is a special web view that focus on controlling states around window. It can be treatedd as
+/// the control panel or navigation bar of the window depending on usages.
+pub struct Panel {
+    id: Option<WebViewId>,
+}
 
-        let size = self.window.available_monitors().nth(0).unwrap().size();
-        let screen = Size2D::new(size.width as i32, size.height as i32);
-        EmbedderCoordinates {
-            hidpi_factor: Scale::new(self.window.scale_factor() as f32),
-            screen,
-            screen_avail: screen,
-            window: (viewport, pos),
-            framebuffer: viewport,
-            viewport: DeviceIntRect::from_origin_and_size(pos, viewport),
-        }
+impl Panel {
+    /// Create a panel from winit window.
+    pub fn new() -> Self {
+        Self { id: None }
```

Comment:
wouldn't it be better, if we generate an ID in constructor (or getting the ID as parameter), and be done with it? Then there is no chance that it could lacking ID at all.

Reply:
in this case `Panel` also could be a tuple struct (if no other parameters are planned, ofc)

Reply:
This is the limitation from Servo. It expect we create a window first to initialize servo. We will have to wait for it to get the ID.